tools/accuracy_checker/accuracy_checker/utils.py

```diff
@@ -518,3 +522,307 @@ def color_format(s, color=Color.PASSED):
+                vheader, next_pos, fd_var = self.read_var_header(fd, endian)
+                varray = self.read_var_array(fd_var, endian, vheader)
+                array[row].append(varray)
+                fd.seek(next_pos)
+        if header['dims'][0] == 1:
+            return self._squeeze(array[0])
+        return self._squeeze(array)
+
+    def _read_struct_array(self, fd, endian, header):
+        field_name_length = self._read_elements(fd, endian, ['miINT32'])
+        if field_name_length > 32:
+            raise ParseError('Unexpected field name length: {}'.format(
+                field_name_length))
+
+        fields = self._read_elements(fd, endian, ['miINT8'], is_name=True)
+        if isinstance(fields, str):
+            fields = [fields]
+
+        empty = lambda: [list() for i in range(header['dims'][0])]
+        array = {}
+        for row in range(header['dims'][0]):
+            for _col in range(header['dims'][1]):
+                for field in fields:
+                    vheader, next_pos, fd_var = self.read_var_header(fd, endian)
+                    data = self.read_var_array(fd_var, endian, vheader)
+                    if field not in array:
+                        array[field] = empty()
+                    array[field][row].append(data)
+                    fd.seek(next_pos)
+        for field in fields:
+            rows = array[field]
+            for i in range(header['dims'][0]):
+                rows[i] = self._squeeze(rows[i])
+            array[field] = self._squeeze(array[field])
+        return array
+
+    def _read_char_array(self, fd, endian, header):
+        array = self._read_numeric_array(fd, endian, header, ['miUTF8'])
+        if header['dims'][0] > 1:
+            array = [self.asstr(bytearray(i)) for i in array]
+        else:
+            array = self.asstr(bytearray(array))
+        return array
+
+    @staticmethod
+    def _squeeze(array):
+        if len(array) == 1:
+            array = array[0]
+        return array
+
+    @staticmethod
+    def _unpack(endian, fmt, data):
+        if fmt == 's':
+            val = struct.unpack(''.join([endian, str(len(data)), 's']),
+                                data)[0]
+        else:
+            num = len(data) // struct.calcsize(fmt)
+            val = struct.unpack(''.join([endian, str(num), fmt]), data)
+            if len(val) == 1:
+                val = val[0]
+        return val
+
+
+def loadmat(filename, meta=False):
+    def eof(fd):
+        b = fd.read(1)
+        end = len(b) == 0
+        if not end:
+            curpos = fd.tell()
+            fd.seek(curpos - 1)
+        return end
+
+    fd = open(filename, 'rb')
+
+    fd.seek(124)
+    tst_str = fd.read(4)
+    little_endian = (tst_str[2:4] == b'IM')
+    endian = ''
+    if (sys.byteorder == 'little' and little_endian) or (sys.byteorder == 'big' and not little_endian):
+        pass
+    elif sys.byteorder == 'little':
+        endian = '>'
+    else:
+        endian = '<'
```

Comment:
I think we can simplify this
```suggestion
    if sys.byteorder == 'little' and not little_endian:
        endian = '>'
    if sys.byteorder == 'big' and little_endian:
        endian = '<'
```